nano/node/election.cpp

```diff
@@ -48,24 +51,168 @@ void nano::election::confirm_once (nano::election_status_type type_a)
 	}
 }
 
-void nano::election::stop ()
+bool nano::election::state_change (nano::election::state_t expected_a, nano::election::state_t desired_a)
```

Comment:
Would be nice if this asserted on invalid state transitions, to catch (admittedly unlikely) bugs where state goes from, say, confirmed to something else.